dump_utils.cpp

```diff
@@ -147,5 +147,79 @@ bool isHostQuiesced()
     }
     return false;
 }
+
+// mapping of severity enum to severity interface
+static std::unordered_map<PelSeverity, std::string> sevMap = {
+    {PelSeverity::INFORMATIONAL,
+     "xyz.openbmc_project.Logging.Entry.Level.Informational"},
+    {PelSeverity::DEBUG, "xyz.openbmc_project.Logging.Entry.Level.Debug"},
+    {PelSeverity::NOTICE, "xyz.openbmc_project.Logging.Entry.Level.Notice"},
+    {PelSeverity::WARNING, "xyz.openbmc_project.Logging.Entry.Level.Warning"},
+    {PelSeverity::CRITICAL, "xyz.openbmc_project.Logging.Entry.Level.Critical"},
+    {PelSeverity::EMERGENCY,
+     "xyz.openbmc_project.Logging.Entry.Level.Emergency"},
+    {PelSeverity::ERROR, "xyz.openbmc_project.Logging.Entry.Level.Error"},
+    {PelSeverity::ALERT, "xyz.openbmc_project.Logging.Entry.Level.Alert"}};
+
+void createPEL(
+    const std::unordered_map<std::string, std::string>& additionalData,
+    const PelSeverity& sev, const std::string& errIntf)
+{
+    try
+    {
+        auto bus = sdbusplus::bus::new_default();
+        constexpr auto loggerObjectPath = "/xyz/openbmc_project/logging";
+        constexpr auto loggerCreateInterface =
+            "xyz.openbmc_project.Logging.Create";
+        constexpr auto loggerService = "xyz.openbmc_project.Logging";
+        std::string pelSeverity =
+            "xyz.openbmc_project.Logging.Entry.Level.Informational";
+        auto itr = sevMap.find(sev);
+        if (itr != sevMap.end())
+            pelSeverity = itr->second;
+
+        sd_bus* pSD_Bus = nullptr;
+        ;
```

Comment:
remove

Reply:
`void Entry::delete_()
{
    auto srcDumpID = sourceDumpId();
    auto dumpId = id;

    if ((!offloadUri().empty()) && (phosphor::dump::isHostRunning()))
    {
        log<level::ERR>(fmt::format("Dump offload in progress, cannot delete "
                                    "dump, id({}) srcdumpid({})",
                                    dumpId, srcDumpID)
                            .c_str());
        elog<sdbusplus::xyz::openbmc_project::Common::Error::Unavailable>();
    }

    auto path = std::filesystem::path(RESOURCE_DUMP_SERIAL_PATH) /
                std::to_string(dumpId);

    log<level::INFO>(
        fmt::format(
            "Resource dump delete id({}) srcdumpid({}) and serial path({})",
            dumpId, srcDumpID, path.string().c_str())
            .c_str());

    // Remove resource dump when host is up by using source dump id
    // which is present in resource dump entry dbus object as a property.
    if ((phosphor::dump::isHostRunning()) && (srcDumpID != INVALID_SOURCE_ID))
    {
        try
        {
            phosphor::dump::host::requestDelete(srcDumpID,
                                                TRANSPORT_DUMP_TYPE_IDENTIFIER);
        }
        catch (const std::exception& e)
        {
            log<level::ERR>(fmt::format("Error deleteing dump from host id({}) "
                                        "host id({}) error({})",
                                        dumpId, srcDumpID, e.what())
                                .c_str());
            elog<sdbusplus::xyz::openbmc_project::Common::Error::Unavailable>();
        }
    }

    // Remove Dump entry D-bus object
    phosphor::dump::Entry::delete_();
    log<level::INFO>(
        fmt::format("Resource dump entry with id({}) is deleted", dumpId)
            .c_str());
    try
    {
        std::filesystem::remove_all(path);
    }
    catch (const std::filesystem::filesystem_error& e)
    {
        // Log Error message and continue
        log<level::ERR>(
            fmt::format("Failed to delete dump file({}), errormsg({})",
                        path.string().c_str(), e.what())
                .c_str());
    }
    // Log PEL for dump /offload
    log<level::INFO>("Log PEL for dump delete or offload");

    std::unordered_map<std::string, std::string> additionalData = {
        {"Dump ID", std::to_string(dumpId)},
        {"Filename", path.string()},
        {"Dump Type", "Resource Dump"}};
    phosphor::dump::createPEL(additionalData,
                              phosphor::dump::PelSeverity::INFORMATIONAL,
                              "xyz.openbmc_project.Dump.Error.Invalidate");
}`
I don't see any sd_bus object in this method to be passed on. Same for system_dump_entry.cpp. Plz correct me if I am wrong.

Reply:
Can't remove as async method needs a sd_bus* object which is not directly available from bus object. Kindly correct me if I am wrong with my analysis.